php/TLS 1.2 Test.php

```diff
@@ -3,57 +3,112 @@
+                                        <hps:ManualEntry>
+                                            <hps:CardNbr>5448144077201692</hps:CardNbr>
+                                            <hps:ExpMonth>02</hps:ExpMonth>
+                                            <hps:ExpYear>2025</hps:ExpYear>
+                                        </hps:ManualEntry>
+                                        <hps:TokenRequest>Y</hps:TokenRequest>
+                                    </hps:CardData>
+                                    <hps:Amt>.01</hps:Amt>
+                                    <hps:AllowDup>Y</hps:AllowDup>
+                                </hps:Block1>
+                            </hps:CreditSale>
+                        </hps:Transaction>
+                    </hps:Ver1.0>
+                </hps:PosRequest>
+            </soapenv:Body>
+        </soapenv:Envelope>
+EOD;
+
+$header = array(
+    'Content-type: text/xml;charset="utf-8"',
+    'Accept: text/xml',
+    'SOAPAction: ""',
+    'Content-length: '.strlen($data),
+);
+function tempStream(){
+    static $tempCounter = 0;
+    return fopen('php://temp' . $tempCounter++, 'w+');
+}
+function readStream($stream){
+    rewind($stream);
+    return stream_get_contents($stream);
+} //iptables -A INPUT -s 12.130.236.166 -j DROP
     $ch = curl_init();
     curl_setopt($ch, CURLOPT_VERBOSE, true);
     $verbose = tempStream();
     curl_setopt($ch, CURLOPT_STDERR, $verbose);
-    curl_setopt($ch, CURLOPT_URL, "https://api2.heartlandportico.com/Hps.Exchange.PosGateway/PosGatewayService.asmx"); //Url together with parameters
-    curl_setopt($ch, CURLOPT_HEADER, 0);
-    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 100);
+    curl_setopt($ch, CURLOPT_URL, $url); //Url together with parameters
+    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 1000);
     curl_setopt($ch, CURLOPT_TIMEOUT, 100);
     curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
-    curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
+    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+    curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
+    if ($data != null) {
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'POST');
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
+    }
+    curl_setopt($ch, CURLOPT_HTTPHEADER, $header);
     curl_setopt($ch, CURLOPT_PROTOCOLS, CURLPROTO_HTTPS);
 
     $curlOut = curl_exec($ch);
-
+    $dom = new DOMDocument;
+    $dom->preserveWhiteSpace = FALSE;
+    $dom->loadXML($data);
+    $dom->formatOutput = TRUE;
+    $data =  str_replace('544814407720', '************', $dom->saveXml()) ;
+    $data =  str_replace($api, substr($api,0,15) . '.......' . substr($api,-5), $data) ;
+    $dom = new DOMDocument;
+    $dom->preserveWhiteSpace = FALSE;
+    $dom->loadXML($curlOut);
+    $dom->formatOutput = TRUE;
+    $curlOut =  $dom->saveXml();
     $curl_getinfo = curl_getinfo($ch);
-
     $curl_errno = curl_errno($ch);
-
     curl_close($ch);
     ?>
 

```

Comment:
More of a verification. Data proves something happened blank leaves you wondering